doc/automated-init.md

```diff
@@ -0,0 +1,113 @@
+# Provisioning Resin.io devices in automated (non-interactive mode)
+
+This document describes how to run the `device init` command in non-interactive mode.
+
+It requires collecting some preliminary information _once_.
+
+The final command to provision the device looks like this:
+
+```bash
+resin device init --app APP_ID --os-version OS_VERSION --drive DRIVE --config "$(cat CONFIG_FILE)" --yes
```

Comment:
We use the full JSON in other places already (like `internal osinit`) and I feel like it's just more flexible. Like if it takes the config as a string you can easily cat it from the file, but not vice versa.